src/core/python_unexposed.h

```diff
@@ -0,0 +1,83 @@
+#include "Python.h"
+
+// Some Python methods that got removed from the public headers in Python 3.13,
+// but don't have immediately obvious replacements. For now, I copied them from
+// the private headers to this file.
+
+// Id methods
+
+static inline PyObject*
+_PyObject_VectorcallMethodId(_Py_Identifier* name,
+                             PyObject* const* args,
+                             size_t nargsf,
+                             PyObject* kwnames)
+{
+  PyObject* oname = _PyUnicode_FromId(name); /* borrowed */
+  if (!oname) {
+    return _Py_NULL;
+  }
+  return PyObject_VectorcallMethod(oname, args, nargsf, kwnames);
+}
+
+static inline PyObject*
+_PyObject_CallMethodIdNoArgs(PyObject* self, _Py_Identifier* name)
+{
+  size_t nargsf = 1 | PY_VECTORCALL_ARGUMENTS_OFFSET;
+  return _PyObject_VectorcallMethodId(name, &self, nargsf, _Py_NULL);
+}
+
+static inline PyObject*
+_PyObject_CallMethodIdOneArg(PyObject* self,
+                             _Py_Identifier* name,
+                             PyObject* arg)
+{
+  PyObject* args[2] = { self, arg };
+  size_t nargsf = 2 | PY_VECTORCALL_ARGUMENTS_OFFSET;
+  assert(arg != NULL);
+  return _PyObject_VectorcallMethodId(name, args, nargsf, _Py_NULL);
+}
+
+extern PyObject*
+_PyObject_CallMethodIdObjArgs(PyObject* obj, _Py_Identifier* name, ...);
+
+extern int
+_PyObject_SetAttrId(PyObject*, _Py_Identifier*, PyObject*);
+
+// Other API stuff
```

Comment:
@vstinner do you have recommended replacements for any of these?

Reply:
Use `PyObject_SetAttr()`. You might continue using `_Py_Identifier` using `_PyUnicode_FromId()`.

Reply:
Makes sense for the Id ones. What about the following ones:
```C
_PyArg_CheckPositional
_PyArg_ParseStack
_PyArg_ParseStackAndKeywords
_PyGen_SetStopIterationValue
_PyGen_FetchStopIterationValue
_Py_HashBytes
_PyObject_NextNotImplemented
_PySet_Update
_PyUnicode_EQ
```

Reply:
`_PyArg_CheckPositional()` is gone, you have to reimplement it using other functions. Or use the internal C API (pycore_xxx.h headers). Or re-export it manually (hack! less reliable):

```
extern int _PyArg_CheckPositional(const char *, Py_ssize_t, Py_ssize_t, Py_ssize_t);
```

Same for other functions.

For `_Py_HashBytes()` and `_PyUnicode_EQ()`, you can use `Py_HashBuffer()` and `PyUnicode_Equal()` from [pythoncapi-compat](https://pythoncapi-compat.readthedocs.io/): they work on Python 3.13 and older.

Reply:
I think we can just wait until we update to Python 3.14 and then switch to using `PyUnicode_Equal()` and `Py_HashBuffer()` since we aren't trying to be source compatible between versions.